core/drand_beacon_control.go

```diff
@@ -1146,27 +1146,29 @@ func (bp *BeaconProcess) StartFollowChain(req *drand.StartFollowRequest, stream
 	cbStore := beacon.NewCallbackStore(ss)
 	defer cbStore.Close()
 
-	syncer := beacon.NewSyncer(bp.log, cbStore, info, bp.privGateway)
-	cb, done := sendProgressCallback(stream, req.GetUpTo(), info, bp.opts.clock, bp.opts.logger)
+	cb, done := sendProgressCallback(stream, req.GetUpTo(), info, bp.opts.clock, bp.log)
 
 	cbStore.AddCallback(addr, cb)
 	defer cbStore.RemoveCallback(addr)
 
-	if err := syncer.Follow(ctx, req.GetUpTo(), peers); err != nil {
-		bp.log.Errorw("", "start_follow_chain", "syncer_stopped", "err", err, "state", "leaving_sync")
-		return err
-	}
+	syncer := beacon.NewSyncManager(&beacon.SyncConfig{
+		Log:    bp.log,
+		Store:  cbStore,
+		Info:   info,
+		Client: bp.privGateway,
+		Clock:  bp.opts.clock,
+	})
+	go syncer.Run()
```

Comment:
is there a reason not to have `Run()` as a private method that's started as a goroutine when calling `NewSyncManager`, vs having it as a separate method that's started as a go routine by the caller immediately after creation?

Reply:
Good point, I don't really have any preference.

Reply:
At least like that, it's explicit that it's running in its own go routine? If you feel I should change it, I can change it. Let me know.